src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -211,5 +205,99 @@ private Socket GetOrCreateAcceptSocket(Socket acceptSocket, bool checkDisconnect
             handle = acceptSocket._handle;
             return acceptSocket;
         }
+
+        private void SendFileInternal(string fileName, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags)
+        {
+            // Open the file, if any
+            FileStream fileStream = null;
+            if (fileName != null && fileName.Length > 0)
+            {
+                fileStream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read);
+            }
```

Comment:
Along with my earlier comments, you could just add a helper to Socket like:
```C#
private static FileStream OpenFile(string name) => !string.IsNullOrEmpty(name) ? File.OpenRead(name) : null;
```
and then use that in all of the places you have this same 5-line snippet.